docs/upgrade/services_removal.md

```diff
@@ -58,6 +58,34 @@ If you use these methods, change their calling appropriately:
   -> `RegistrationMail::createMessage(MailTemplate $mailTemplate, $user)`
 - `OrderMailService::getMessageDataByOrder(Order $order, MailTemplate $mailTemplate)`  
   -> `OrderMail::createMessage(MailTemplate $mailTemplate, $order)`
+- `CustomerPasswordService::resetPassword(User $user)`  
+  -> `User::resetPassword(HashGenerator $hashGenerator)`
+- `CustomerPasswordService::isResetPasswordHashValid(User $user, $hash)`  
+  -> `User::isResetPasswordHashValid(?string $hash): bool`
+- `CustomerPasswordService::changePassword(User $user, $password)`  
+  -> `User::changePassword(EncoderFactoryInterface $encoderFactory, $password)`
+- `CustomerPasswordService::setNewPassword(User $user, $hash, $newPassword)`  
+  -> `User::setNewPassword(EncoderFactoryInterface $encoderFactory, ?string $hash, string $newPassword)`
+- `ProductVariantService::checkProductIsNotMainVariant(Product $product)`  
+  -> `Product::checkIsNotMainVariant()`
+- `ProductVariantService::refreshProductVariants(Product $mainProduct, array $currentVariants)`  
+  -> `Product::refreshProductVariants(array $currentVariants)`
```

Comment:
maybe "refreshVariants" would be better, however, you would need to change it in e118616801800fdb776aaac0f729f7da2e513932 as well. I do not insist on the renaming, it is up to you :wink: